### PR TITLE
Added Slack#set_topic

### PR DESCRIPTION
**Pull request by jnalley**

### CODE REVIEW

**Comment by jimmycuadra** [comment]
Awesome! Thanks for doing this work. I'll add inline comments for a couple of small style points.

File: spec/lita/adapters/slack/api_spec.rb [hunk 1]
```diff
@@ -66,6 +66,60 @@
     end
   end
 
+  describe "#set_topic" do
+    let(:channel) { 'C1234567890' }
+    let(:topic) { 'Topic' }
+    let(:stubs) do
+      Faraday::Adapter::Test::Stubs.new do |stub|
+        stub.post('https://slack.com/api/channels.setTopic', token: token,
```
Comment (jimmycuadra) [inline spec/lita/adapters/slack/api_spec.rb]: Preferred indentation for multi-line method calls:

``` ruby
foo(
  bar,
  baz,
)
```

File: spec/lita/adapters/slack/api_spec.rb [hunk 2]
```diff
@@ -66,6 +66,60 @@
...
+                  channel: channel, topic: topic) do
+          [http_status, {}, http_response]
+        end
+      end
+    end
+
+    describe "with a successful response" do
```
Comment (jimmycuadra) [inline spec/lita/adapters/slack/api_spec.rb]: Use `context` instead of `describe` for these nested blocks.

**Comment by jnalley** [comment]
I think I have fixed the issues you highlighted.

**Comment by jimmycuadra** [comment]
Thank you! Released in v1.4.0.

**Pull request merged**